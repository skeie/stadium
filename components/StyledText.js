// @flow

import { Text, StyleSheet } from 'react-native';
import * as React from 'react';

export const styles = StyleSheet.create({
    style: { color: '#553555', fontFamily: 'poppins' },
    bold: { color: '#553555', fontFamily: 'poppins-bold' },
    header: { color: '#553555', fontSize: 20, fontFamily: 'poppins' },
    subHeader: { color: '#553555', fontSize: 18, fontFamily: 'poppins' },
});

type MonoProps = {
    style?: number | Object,
};

type Type = 'style' | 'header' | 'subHeader' | 'bold';

type PoppinsProps = MonoProps & {
    children: React.Node,
    type?: Type,
    style?: number | Object,
};

export const MonoText = (props: MonoProps) => {
    return <Text {...props} style={[props.style, { fontFamily: 'space-mono' }]} />;
};

export const Poppins = (props: PoppinsProps) => {
    const { children } = props;
    return (
        <Text
            ellipsizeMode="tail"
            {...props}
            style={[styles.style, props.type && styles[props.type], props.style]}>
            {children}
        </Text>
    );
};
