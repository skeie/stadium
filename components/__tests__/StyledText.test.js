import 'react-native';
import React from 'react';
import { MonoText, Poppins } from '../StyledText';
import renderer from 'react-test-renderer';

it('Mono renders correctly', () => {
  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();

  expect(tree).toMatchSnapshot();
});
it('Poppins renders correctly', () => {
  const tree = renderer.create(<Poppins>Snapshot test!</Poppins>).toJSON();

  expect(tree).toMatchSnapshot();
});
