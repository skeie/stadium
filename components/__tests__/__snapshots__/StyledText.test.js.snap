// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mono renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`Poppins renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#553555",
        "fontFamily": "poppins",
      },
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
